src/helm/proxy/clients/lit_gpt_client.py

```diff
@@ -0,0 +1,179 @@
+    ):
+        torch.set_float32_matmul_precision("high")
+
+        if strategy == "fsdp":
+            strategy = FSDPStrategy(auto_wrap_policy={Block}, cpu_offload=False)
+        fabric = L.Fabric(devices=devices, accelerator=device, precision=precision, strategy=strategy)
+        fabric.launch()
+        logger.info("Using device: {}".format(fabric.device))
+
+        checkpoint_dir = Path(checkpoint_dir)
+        check_valid_checkpoint_dir(checkpoint_dir)
+
+        with open(checkpoint_dir / "lit_config.json") as fp:
+            config = Config(**json.load(fp))
+
+        checkpoint_path = checkpoint_dir / "lit_model.pth"
+        logger.info(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
+        with fabric.init_module(empty_init=True), quantization(quantize):
+            model = GPT(config)
+
+        with lazy_load(checkpoint_path) as checkpoint:
+            model.load_state_dict(checkpoint, strict=quantize is None)
+
+        model.eval()
+        self.model = fabric.setup(model)
+        self.tokenizer = Tokenizer(checkpoint_dir)
+        self.fabric = fabric
+
+
+class LitGPTClient(Client):
+    """Client for evaluating Lit-GPT (from Lightning AI) supported LLMs"""
+
+    def __init__(
+        self,
+        cache_config: CacheConfig,
+        checkpoint_dir: str = "",
+        precision: str = "bf16-true",
+        device="auto",
+        devices: int = 1,
+        strategy: str = "auto",
+        quantize: Optional[Literal["bnb.nf4", "bnb.nf4-dq", "bnb.fp4", "bnb.fp4-dq", "bnb.int8", "gptq.int4"]] = None,
+    ):
+        self.cache = Cache(cache_config)
+        lit_gpt = LitGPT(checkpoint_dir, precision, device, devices, strategy, quantize)
+        self.model = lit_gpt.model
+        self.tokenizer = lit_gpt.tokenizer
+        self.fabric = lit_gpt.fabric
+
+    def make_request(self, request: Request) -> RequestResult:
+        model = self.model
+        tokenizer = self.tokenizer
+        fabric = self.fabric
+        encoded = tokenizer.encode(request.prompt, bos=True, eos=False, device=fabric.device)
+        prompt_length = encoded.size(0)
+        max_returned_tokens = prompt_length + request.max_tokens
+        assert max_returned_tokens <= model.config.block_size, (
+            max_returned_tokens,
+            model.config.block_size,
+        )  # maximum rope cache length
+
+        model.clear_kv_cache()
+
+        with fabric.init_tensor():
+            # set the max_seq_length to limit the memory usage to what we need
+            model.max_seq_length = max_returned_tokens
+
+        t0 = time.perf_counter()
+        # helm doesn't have anything equivalent to top_k at the moment
+        # TODO: allow temperature=0, pick the top token rather than sampling.
+        stop_tokens: List[torch.int] = [tokenizer.encode(e, device=fabric.device) for e in request.stop_sequences]
```

Comment:
Type annotation should be `List[torch.Tensor]`.

Reply:
updated